### PR TITLE
Update ReorderableListView docs to avoid confusion between desktop and mobile

### PR DESCRIPTION
**Pull request by idkq**
Prevents this https://github.com/flutter/flutter/issues/80553

### CODE REVIEW

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 1]
```diff
@@ -29,6 +29,11 @@ import 'theme.dart';
 /// The [onReorder] parameter is required and will be called when a child
 /// widget is dragged to a new position.
 ///
+/// Note: On desktop platforms, a drag handle is stacked over the center of
```
[the remark this answers is not in the document]
Comment (idkq) [reply packages/flutter/lib/src/material/reorderable_list.dart]: Done

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 2]
```diff
@@ -29,6 +29,11 @@ import 'theme.dart';
 /// The [onReorder] parameter is required and will be called when a child
 /// widget is dragged to a new position.
 ///
+/// By default, a drag handle is stacked over the center of each item's trailing
+/// edge on desktop platforms only; on mobile platforms, no handle is visible
+/// and a long press anywhere on the item starts a drag. To customize this
+/// behavior check [ReorderableListView.buildDefaultDragHandles].
```
Comment (Piinks) [inline packages/flutter/lib/src/material/reorderable_list.dart:35]:
```suggestion
/// edge on [TargetPlatformVariant.desktop] platforms. On [TargetPlatformVariant.mobile] platforms, no handle is visible
/// and a long press anywhere on the item starts a drag. To change this
/// behavior use [ReorderableListView.buildDefaultDragHandles].
```

**Review by Piinks — COMMENTED** [review]
Hey @idkq thanks for the contribution!

**Review by Piinks — APPROVED** [review]
LGTM, thank you! I am reaching out to @Hixie for test exemption. :)

File: packages/flutter/lib/src/material/reorderable_list.dart (hunk 2 above)
Comment (darrenaustin) [reply packages/flutter/lib/src/material/reorderable_list.dart]: While I like this wording, it currently isn't true 😄.  The reorderable list treats `TargetPlatform.fuchsia` as a desktop platform, but the `TargetPlatformVariant` puts it in the `mobile` category. We should fix the reorderable list view to treat it as mobile as well to be consistent.

That said, we should go ahead with this and I will take care of fixing the issue in https://github.com/flutter/flutter/issues/84376.

**Review by darrenaustin — CHANGES_REQUESTED** [review]
@idkq thanks for the contribution. Let's go with @Piinks suggested wording and then we can land this.

**Review by darrenaustin — APPROVED** [review]
Oops. I was ninja'd. LGTM too.

**Comment by skia-gold** [comment]
Gold has detected about 3 new digest(s) on patchset 3.
View them at https://flutter-gold.skia.org/cl/github/80576

**Comment by Hixie** [comment]
test-exempt: only affects comments

**Comment by Piinks** [comment]
Hey @idkq can you rebase this with the latest from master? I think the age of this PR is causing some issues with CI.

**Comment by idkq** [comment]
@Piinks I might need to open a new PR as this did not work - not sure what is going on

**Comment by darrenaustin** [comment]
You should just be able to rebase your change on top of the changes from master with something like:

```sh
git pull --rebase upstream master 
git push --force origin idkq:patch-1
```

But if you have already tried that and it didn't work then go ahead and open a new PR with your changes.

**Comment by idkq** [comment]
Apologies for the delay. @darrenaustin I tried that but did not work. I opened a new one here https://github.com/flutter/flutter/pull/86737

**Pull request closed without merging**